src/content/en/tools/workbox/guides/migrate.md

```diff
@@ -0,0 +1,224 @@
+[supported](https://github.com/GoogleChromeLabs/sw-precache#command-line-interface) configuring via
+either command line flags or a configuration file, the `workbox` CLI requires that all configuration
+options be provided in a configuration file, using [CommonJS
+`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports).
+
+The `workbox` CLI supports a number of different modes (use `workbox --help` to see all of them),
+but the mode that most closely matches `sw-precache`'s functionality is `generateSW`. So a call to
+
+```
+$ sw-precache --config='sw-precache-config.js'
+```
+
+can be expressed as
+
+```
+$ workbox generateSW workbox-config.js
+```
+
+### From the sw-precache node module to the workbox-build node module
+
+Developers using the `node` API for `sw-precache`, either as part of a `gulp`/`Grunt` workflow or
+just within a custom `node` build script, can migrate by switching to the `workbox-build` `node`
+module.
+
+`workbox-build`'s `generateSW()` function most closely matches the `sw-precache` `node` module's
+`write()` function. One key difference is that `generateSW()` always returns a Promise, while the
+old `write()` function supported both a callback and Promise-based interface.
+
+`gulp` usage along the lines of
+
+```
+const swPrecache = require('sw-precache');
+gulp.task('generate-service-worker', function() {
+  return swPrecache.write('service-worker.js', {
+    // Config options.
+  });
+});
+```
+
+can be changed to
+
+```
+const workboxBuild = require('workbox-build');
+gulp.task('generate-service-worker', function() {
+  return workboxBuild.generateSW({
+    // Config options.
+  });
+});
+```
+
+### From the sw-precache-webpack-plugin to the Workbox webpack plugin
+
+Developers using the
+[`sw-precache-webpack-plugin`](https://www.npmjs.com/package/sw-precache-webpack-plugin) as part of
+their [webpack](https://webpack.js.org/) build process can migrate by switching to the `GenerateSW`
+class within the `workbox-webpack-plugin` module.
+
+`workbox-webpack-plugin` integrates directly with the webpack build process and "knows" about all of
+the assets generated by the a given compilation. This means that, for many use cases, you can rely
+on the default behavior of `workbox-webpack-plugin` without additional configuration, and get an
+equivalent service worker to what `sw-precache-webpack-plugin` provides.
+
+```
+const SWPrecacheWebpackPlugin = require('sw-precache-webpack-plugin');
+const webpackConfig = {
+  // ...
+  plugins: [
+    new SWPrecacheWebpackPlugin({
+        dontCacheBustUrlsMatching: /\.\w{8}\./,
+        filename: 'service-worker.js',
+      }
+    ),
+  ],
+};
+```
+
+can be changed to
+
+```
+const {GenerateSW} = require('workbox-webpack-plugin');
+const webpackConfig = {
+  // ...
+  plugins: [
+    new GenerateSW({
+      // Config options, if needed.
+    }),
+  ],
+};
+```
+
+## sw-toolbox migrations
+
+### Migrate from hand-crafted sw-toolbox usage to workbox-sw
```

Comment:
I don't think the word 'usage' adds anything to the meaning of this phrase. Also shorter titles are better for small screens.